packages/caliper-fabric/test/connector-configuration/ConnectionProfileDefinition.js

```diff
@@ -436,4 +436,460 @@ describe('A Connection Profile Definition', async () => {
+                        },
+                    }
+                }
+            };
+            const connectionProfileDefinition = new ConnectionProfileDefinition(mspId, {
+                loadedConnectionProfile: invalidPemConnectionProfile,
+                discover: false
+            });
+            await connectionProfileDefinition.getTlsCertForPeer(peer).should.be.rejectedWith(
+                `pem provided for ${peer} in the connection profile .tlsCACerts.pem is not valid`
+            );
+        });
+        it('should throw an error if path provided through tlsCAcets.path does not point to a valid cert', async () => {
+            const invalidPemConnectionProfile = {
+                peers: {
+                    'peer0.org1.example.com': {
+                        tlsCACerts: {
+                            path:'.test/sample-configs/invalid.pem'                        },
+                    }
+                }
+            };
+            const path = CaliperUtils.resolvePath(invalidPemConnectionProfile.peers[peer].tlsCACerts.path);
+            const connectionProfileDefinition = new ConnectionProfileDefinition(mspId, {
+                loadedConnectionProfile: invalidPemConnectionProfile,
+                discover: false
+            });
+            await connectionProfileDefinition.getTlsCertForPeer(peer).should.be.rejectedWith(
+                `path property does not point to a file that exists at ${path} for ${peer}`
+            );
+        });
+    });
+
+
+    describe('when getting the grpc endpoint for a peer', () => {
+
+        const peer = 'peer0.org1.example.com';
+
+        it('should return the parsed for a grpcs url', async () => {
+            const grpcsConnectionProfile = {
+                peers: {
+                    'peer0.org1.example.com': {
+                        url: 'grpcs://localhost:7051'
+                    }
+                }
+            };
+            const connectionProfileDefinition = new ConnectionProfileDefinition(mspId, {
+                loadedConnectionProfile: grpcsConnectionProfile,
+                discover: true
+            });
+            connectionProfileDefinition.getGrpcEndPointForPeer(peer).should.deep.equal('localhost:7051');
+        });
+
+        it('should return the parsed url for a grpc url', async () => {
```

Comment:
We can change this description to be similar to above `should return the correct endpoint for a grpc url`